sw/banshee/src/tran.rs

```diff
@@ -2119,6 +2119,36 @@ impl<'a> InstructionTranslator<'a> {
             riscv::OpcodeRdRs1Rs2::Or => LLVMBuildOr(self.builder, rs1, rs2, name),
             riscv::OpcodeRdRs1Rs2::Xor => LLVMBuildXor(self.builder, rs1, rs2, name),
             riscv::OpcodeRdRs1Rs2::Mul => LLVMBuildMul(self.builder, rs1, rs2, name),
+            riscv::OpcodeRdRs1Rs2::Mulhu => {
+                let tmp = LLVMBuildMul(
+                    self.builder,
+                    LLVMBuildZExt(self.builder, rs1, LLVMInt64Type(), NONAME),
+                    LLVMBuildZExt(self.builder, rs2, LLVMInt64Type(), NONAME),
+                    name,
+                );
+                let tmp = LLVMBuildLShr(
+                    self.builder,
+                    tmp,
+                    LLVMConstInt(LLVMInt64Type(), 32 as u64, 0),
+                    NONAME,
+                );
+                LLVMBuildTrunc(self.builder, tmp, LLVMInt32Type(), NONAME)
+            }
+            riscv::OpcodeRdRs1Rs2::Mulh => {
+                let tmp = LLVMBuildMul(
+                    self.builder,
+                    LLVMBuildSExt(self.builder, rs1, LLVMInt64Type(), NONAME),
+                    LLVMBuildSExt(self.builder, rs2, LLVMInt64Type(), NONAME),
+                    name,
+                );
+                let tmp = LLVMBuildLShr(
+                    self.builder,
+                    tmp,
+                    LLVMConstInt(LLVMInt64Type(), 32 as u64, 0),
+                    NONAME,
+                );
+                LLVMBuildTrunc(self.builder, tmp, LLVMInt32Type(), NONAME)
+            }
```

Comment:
Nice. Could we add `mulhsu` as well?
```suggestion
            }
            riscv::OpcodeRdRs1Rs2::Mulhsu => {
                let tmp = LLVMBuildMul(
                    self.builder,
                    LLVMBuildSExt(self.builder, rs1, LLVMInt64Type(), NONAME),
                    LLVMBuildZExt(self.builder, rs2, LLVMInt64Type(), NONAME),
                    name,
                );
                let tmp = LLVMBuildLShr(
                    self.builder,
                    tmp,
                    LLVMConstInt(LLVMInt64Type(), 32 as u64, 0),
                    NONAME,
                );
                LLVMBuildTrunc(self.builder, tmp, LLVMInt32Type(), NONAME)
            }
```

Reply:
could you also write a test for it?